### DISCUSSION TITLE
Upgrading Pillow from version 8.3.2-> 9.1.1

### DISCUSSION THREAD

**Discussion by sambidsresh** [python-pillow/Pillow#6356] [Q&A] [answered]
I am upgrading pillow from version 8.3.2 to 9.1.1.
A little confused by the release notes. version 9.0.0 and above are built against libjpeg-turbo.
Does it mean we must have libjpeg-turbo installed on our servers as well?
or is it fine to stick with libjpeg-dev package we have.

**Comment by nulano** [comment, marked as answer]
Depends on how you install Pillow.

If it is installed from binary wheel from PyPI, you don't need either libjpeg or libjpeg-turbo, it is bundled in the wheel. (this is the change in the release notes)

If it is installed from source, you can use either libjpeg-dev or libjpegturbo-dev.

If it is installed from a system package, it should resolve dependencies itself. In this case it is up to the package maintainers to choose libjpeg or libjpeg-turbo.
